pkg/crd/init.go

```diff
@@ -143,6 +144,23 @@ func tagToColumn(f reflect.StructField) (apiextv1.CustomResourceColumnDefinition
 	return c, true
 }
 
+type openAPISchemaProvider interface {
+	OpenAPISchemaType() []string
+	OpenAPISchemaFormat() string
+}
+
+func openAPISchemaKind(t reflect.Type) reflect.Kind {
+	if o, ok := reflect.New(t).Interface().(openAPISchemaProvider); ok {
+		if st := o.OpenAPISchemaType(); len(st) > 0 {
```

Comment:
I looked through the core Kubernetes codebase and couldn't find anything that returned anything other than just `["string"`] - even `intOrString` just returns `["string"]`

* https://github.com/kubernetes/apimachinery/blob/master/pkg/util/intstr/intstr.go#L133
* https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L405
* https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go#L168
* https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/micro_time.go#L167
* https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/duration.go#L61

Reply:
Okay this should be fine, I would still rather only accept a slice of len == 1 since structs outside of ones defined in upstream can also hit this, but since this isn't a library used by people outside of Rancher this should be fine.